sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala

```diff
@@ -67,6 +68,27 @@ class SQLAppStatusListener(
     }
   }
 
+  // visible for tests
+  private[spark] def recoverLiveEntities(): Unit = {
+    if (!live) {
+      kvstore.view(classOf[SQLExecutionUIData])
+        // null metricValues potentially indicates a live SQLExecutionUIData
```

Comment:
Is it exhaustive? or there are some cases we can't handle? Is there any way to make this deterministic?

Reply:
You have a good point here. Condition of `null metricValues` seems to be flaky. I'll check `SQLExecutionUIData`'s `completionTime` and all jobs status within `SQLExecutionUIData` to see whether it's still live, which seems more reliable. WDYT?